src/fs.rs

```diff
@@ -151,6 +155,141 @@ pub fn create_workspace_archive(working_root_dir: &str, execution_id: &str) -> R
+    let mut file = match File::create(&path) {
+        Err(e) => return Err(e),
+        Ok(file) => file,
+    };
+
+    match file.write(content.as_bytes()) {
+        Err(e) => Err(e),
+        Ok(_) => {
+            file.flush()?;
+            Ok(path
+                .to_str()
+                .map(|e| e.to_string())
+                .ok_or_else(|| Error::from(ErrorKind::NotFound))?)
+        }
+    }
+}
+
+pub fn remove_lines_starting_with(path: String, starter: &str) -> Result<String, std::io::Error> {
+    info!("editing yaml backup file {}", path);
+
+    let file = OpenOptions::new()
+        .read(true)
+        .open(path.as_str())
+        .map_err(|_| Error::from(ErrorKind::NotFound))?;
+
+    let content = BufReader::new(file.try_clone().unwrap())
+        .lines()
+        .filter(|line| !line.as_ref().unwrap().contains(starter))
+        .map(|line| line.unwrap())
+        .collect::<Vec<String>>()
+        .join("\n");
+
+    let mut file = OpenOptions::new()
+        .write(true)
+        .truncate(true)
+        .open(path.as_str())
+        .map_err(|_| Error::from(ErrorKind::NotFound))?;
+
+    match file.write(content.as_bytes()) {
+        Err(e) => Err(e),
+        Ok(_) => Ok(path),
+    }
+}
+
+pub fn list_yaml_backup_files<P>(working_root_dir: P) -> Result<Vec<String>, std::io::Error>
+where
+    P: AsRef<Path>,
+{
+    let files = WalkDir::new(working_root_dir)
+        .follow_links(true)
+        .into_iter()
+        .filter_map(|e| e.ok());
+    let mut backup_paths: Vec<String> = vec![];
+    for file in files {
+        if file
+            .file_name()
+            .to_str()
+            .ok_or_else(|| Error::from(ErrorKind::NotFound))?
+            .to_string()
+            .contains("-q-backup.yaml")
+        {
+            backup_paths.push(file.path().to_str().expect("No file path").to_string())
+        }
+    }
+
+    if backup_paths.is_empty() {
+        return Err(Error::from(ErrorKind::NotFound));
+    }
+
+    Ok(backup_paths)
+}
+
+pub fn create_yaml_file_from_secret<P>(working_root_dir: P, secret: SecretItem) -> Result<String, CommandError>
+where
+    P: AsRef<Path>,
+{
+    let message = format!("Unable to decode secret {}", secret.metadata.name.clone());
+    let secret_data = secret.data.values().next();
+    let secret_content = match secret_data.is_some() {
+        true => secret_data.unwrap().to_string(),
+        false => return Err(CommandError::new(message.clone(), Some(message), None)),
+    };
+
+    let content = match decode(secret_content) {
+        Ok(bytes) => from_utf8_lossy(&bytes[1..bytes.len() - 1]).to_string(),
+        Err(_) => return Err(CommandError::new(message.clone(), Some(message), None)),
+    };
+    match create_yaml_backup_file(working_root_dir.as_ref(), secret.metadata.name.clone(), None, content) {
+        Ok(path) => Ok(path),
+        Err(e) => {
+            let message = format!("Unable to create backup file from secret {}: {}", secret.metadata.name, e);
```

Comment:
Please don't do that, it can leak stuff. Make sure safe message is safe and do not put error message directly into the safe message.

```
Err(CommandError::new(
  format!("Unable to create backup file from secret {}.", secret.metadata.name), 
  Some(e.to_string()),
  None))
```